api/openapi-spec/swagger.json

```diff
@@ -248,6 +256,40 @@
         }
       }
     },
+    "io.argoproj.workflow.v1alpha1.GCSArtifact": {
+      "description": "GCSArtifact is the location of a GCS artifact",
+      "required": [
+        "bucket",
+        "credentialsSecret",
```

Comment:
Not sure if this PR is still active, but just in case: Could `credentialsSecret` be made optional here? I imagine many teams wanting to use GCS as an artifact store will be running on GKE. Best-practice in GKE is now to use [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) for authentication instead of mounting service account keys in secrets.